Added automated way to build documentation and deploy it to github pages

fedem-p opened the pull request:
After some bugs, it can finally be triggered only by a push from the main branch.

vigji commented:
This is absolutely great @fedem-p!!!

There seems to be a funny duplications of titles in the left, do you know why that happens?

fedem-p commented:
I reorganized everything and now it should display correctly the different sections +  the API.
 
todo:
- link the readme instead of copying ( I tried a bit but it seems harder than I though..    using rst to link it seems to not work with jupyter books, and I couldn't find any solution using only markdown)
-  add an index on the homepage (?)

diegoasua commented:
> 
> 
> I reorganized everything and now it should display correctly the different sections + the API.
> 
> todo:
> 
>     * link the readme instead of copying ( I tried a bit but it seems harder than I though..    using rst to link it seems to not work with jupyter books, and I couldn't find any solution using only markdown)
> 
>     * add an index on the homepage (?)

For the first part, does [this](https://github.com/miyakogi/m2r/issues/41) work? For the second part, there is already a searchable index on the left side when checking the compiled web, and as adding the index to homepage seems to be messy it is probably in your interest to just forget about that.

fedem-p commented:
> > I reorganized everything and now it should display correctly the different sections + the API.
> > todo:
> > ```
> > * link the readme instead of copying ( I tried a bit but it seems harder than I though..    using rst to link it seems to not work with jupyter books, and I couldn't find any solution using only markdown)
> > 
> > * add an index on the homepage (?)
> > ```
> 
> For the first part, does [this](https://github.com/miyakogi/m2r/issues/41) work? For the second part, there is already a searchable index on the left side when checking the compiled web, and as adding the index to homepage seems to be messy it is probably in your interest to just forget about that. I like the new style of the docs, it feels very clean

diegoasua commented on .github/workflows/doc_deploy.yml:
```diff
@@ -0,0 +1,35 @@
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
```
Is there a need to install all [requirements](https://github.com/portugueslab/sashimi/blob/master/requirements.txt) from the main program, to build the docs? Is just toml enough maybe? Or not even, in which case `Install dependencies` job can be skipped altogether.

fedem-p replied:
You were right! Thanks!

diegoasua commented on docs/intro.md:
```diff
@@ -0,0 +1,94 @@
+# Sashimi
+
+<a href="url"><img 
+src="https://github.com/portugueslab/sashimi/blob/master/sashimi/icons/main_icon.png"
```
[This image](https://portugueslab.github.io/sashimi/intro.html) does not seem to load correctly. Try without the hyperlink (`<a></a>`) tag:

```html
<img src=https://github.com/portugueslab/sashimi/blob/master/sashimi/icons/main_icon.png align="left" height="190" width="270">
```

diegoasua commented on docs/intro.md:
```diff
@@ -0,0 +1,94 @@
+
+    conda activate sashimi
+    
+After this you moght have to install two extra dependencies for controlling a Cobolt laser:
```
Technically this should read for controlling a laser through serial communication, such as Cobolt lasers.

diegoasua reviewed (approved):
Awesome work on the sphinx automation in `docs/_templates`. Overall I don't have anything to add and the very few comments on this are just picky things on my side. After they have been addressed feel free to merge to main branch. Good work! 😃

vilim commented on docs/index.md:
```diff
@@ -0,0 +1,4 @@
+# Welcome to the Sashimi documentation!
+
+maybe add index here?
```
It should be an internal comment for us, not user-visible

vilim reviewed (approved):
Only one minor comment, otherwise good to merge!

fedem-p commented:
I've fixed the problem with the image by changing the URL, now with this, it works both on GitHub and the documentation.
I've also restyled the structure a tiny bit and added an index at the home page (that can be removed easily if you don't like it)

Regarding linking the README file directly, I couldn't find anything that worked.  
The problem seems to be the fact that I cannot use the mainstream way to do it with an .rst because it doesn't compile properly with Jupiter books.  There is some workaround to do that just with .md files, but I still need to find one that works.  
In the end, I just copied pasted the content and adjusted the structure to have different sections for installation and configuration.

diegoasua commented:
And.... Done!
